include/cantera/numerics/Func1.h

```diff
@@ -265,6 +274,68 @@ class Pow1 : public Func1
     virtual Func1& derivative() const;
 };
 
+
+//! implements a tabulated function
+class Tabulated1 : public Func1
+{
+public:
+    Tabulated1(const std::vector<double>& tvec, const std::vector<double>& fvec,
+               const std::string& method = "linear") :
+        Func1() {
+        if (tvec.size() != fvec.size()) {
+            throw CanteraError("Tabulated1::Tabulated1",
+                               "sizes of vectors do not match ({} vs {}).",
+                               tvec.size(), fvec.size());
+        } else if (tvec.empty()) {
+            throw CanteraError("Tabulated1::Tabulated1",
+                               "vectors must not be empty.");
+        }
+        m_tvec = tvec;
+        m_fvec = fvec;
+        if (method == "linear") {
+            m_isLinear = true;
+        } else if (method == "previous") {
+            m_isLinear = false;
+        } else {
+            throw CanteraError("Tabulated1::Tabulated1",
+                               "interpolation method '{}' is not implemented",
+                               method);
+        }
+    }
+
+    Tabulated1(const Tabulated1& b) :
+        Func1(b) {
+    }
+
+    Tabulated1& operator=(const Tabulated1& right) {
+        if (&right == this) {
+            return *this;
+        }
+        Func1::operator=(right);
+        return *this;
+    }
+
+    virtual std::string write(const std::string& arg) const;
+    virtual int ID() const {
+        return TabulatedFuncType;
+    }
+    virtual double eval(double t) const;
+    virtual Func1& duplicate() const {
+        if (m_isLinear) {
+            return *(new Tabulated1(m_tvec, m_fvec, "linear"));
+        } else {
+            return *(new Tabulated1(m_tvec, m_fvec, "previous"));
+        }
+    }
+
+    virtual Func1& derivative() const;
+private:
+    std::vector<double> m_tvec;
+    std::vector<double> m_fvec;
```

Comment:
```suggestion
    vector_fp m_fvec;
```
Here and anywhere else this appears.